src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1029,6 +1030,14 @@ def _validate_vm_nic_delete_option(namespace):
                                             "specify delete option for the new nic created: --nic-delete-option Delete")
 
 
+def _validate_vm_data_disk_delete_option(namespace):
+    if "attach_data_disks" not in namespace:
+        return
+    if not namespace.attach_data_disks and namespace.data_disk_delete_option:
```

Comment:
Please add a variable to save the result of judgment `attach_data_disks` for easy understanding